src/core/nginx.go

```diff
@@ -316,7 +317,11 @@ func hasConfPath(files []*proto.File, confPath string) bool {
 }
 
 func (n *NginxBinaryType) WriteConfig(config *proto.NginxConfig) (*sdk.ConfigApply, error) {
-	log.Tracef("Writing config: %+v\n", config)
+	if log.IsLevelEnabled(log.TraceLevel) {
+		jsonConfig, _ := json.Marshal(config)
```

Comment:
I think it make sense to handle/check the error returned by `json.Marshal`?

Reply:
It makes sense, yes. I've changed the code to log the raw message if the marshal operation fails.